src/main/java/org/apache/sling/caconfig/ConfigurationBuilder.java

```diff
@@ -86,4 +86,15 @@
      */
     @NotNull <T> Collection<T> asAdaptableCollection(@NotNull Class<T> clazz);
 
+    /**
+     * Check if the configuration node is existing in the JCR. This method checks for the configuration existence based on {@code configName} defined in the
+     * configuration definition. It checks if the configuration is available for the context path in the current resource hierarchy and if not found, also
+     * checks in the global fall-back paths configured via {@link org.apache.sling.caconfig.resource.impl.def.DefaultConfigurationResourceResolvingStrategy}
+     * which by default are {@code /conf/global}, {@code /apps/config} and {@code /libs/config}.This method does not consider the default values provided in
+     * the configuration definition.
+     * @param clazz Class that can be adapted from a {@link org.apache.sling.api.resource.Resource}
+     * @param <T> Annotation class type
+     * @return True/False based on configuration resource node existence in JCR.
+     */
+    @NotNull <T> boolean has(@NotNull Class<T> clazz);
```

Comment:
this method only supports the use case of using a configuration annotation class with the ConfigurationBuilder.
but it is also possible to use the ConfigurationBuilder without such a class - by using the asValueMap, asValueMapCollection, asAdaptable methods, as adaptableCollection - all these methods required a name to be given by the "name" method.
so we need a "has" variant as well for those use cases - either by adding a "has()" method which required a name given before, or by adding a has(String) method which accepts the name. not sure what is the better usage pattern.

Reply:
@stefanseifert IMHO `asValueMap`, `asValueMapCollection`, `asAdaptable` and  `sAdaptableCollection` all return either `null` or an empty value map in case there is no config. Therefore, I don't see the need to add other `has` methods.

Reply:
@kwin I think it returns the default value in that case as well. for example, 
`configurationBuilder.name("/project/tenantA/caconfig").asValueMap().isEmpty()`  considers the default value if it's in the config definition class.

Reply:
yes, for example in case of default values or override values they are also returned in the valuemap.

Reply:
Should we then go with one method for every case? has(String configName) What do you suggest @kwin

Reply:
I have refactored this method and now it expects a `configName ` parameter for all the cases.

Reply:
The problem is that you cannot easily get the configName from the annotation (as the method for extracting it is not public API), so I would strongly suggest to have two different `has` methods (one taking a string configName and the other one taking the annotation class).